clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -86,11 +86,21 @@
     List<ClientResponse> poll(long timeout, long now);
 
     /**
+     * Diconnects the connection to a particular node, if there is one.
+     * Any pending ClientRequests for this connection will receive disconnections.
+     *
+     * @param nodeId The id of the node
+     */
+    void disconnect(String nodeId);
+
+    /**
      * Closes the connection to a particular node (if there is one).
+     * All requests on the connection will be cleared.  ClientRequest callbacks will not be invoked
+     * for the cleared requests, nor will they be returned from poll().
      *
      * @param nodeId The id of the node
      */
-    void close(String nodeId);
+    void forget(String nodeId);
```

Comment:
Do we really need this rename? `NetworkClient.close()` is consistent with `Selector.close()` that doesn't populate the `disconnected` list. I'd rather keep this as `close()` for now. We can reconsider later.